lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -54,11 +54,24 @@ run = do
   pkgs <- getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> do
     freePort <- getFreePort
-    assets <- withProjectRoot "." $ \root ->
-      -- `--raw` is not available with old nix-instantiate. It drops quotation
-      -- marks and trailing newline, so is very convenient for shelling out.
-      readProcessAndLogStderr Debug $
-        proc "nix" ["eval", "-f", root, "passthru.staticFilesImpure", "--raw"]
+    assets <- withProjectRoot "." $ \root -> do
+      let importableRoot = if "/" `isInfixOf` root
+            then root
+            else "./" <> root
+      isDerivation <- readProcessAndLogStderr Debug $
+        proc "nix"
+          [ "eval"
+          , "-f"
+          , root
+          , "(let a = import " <> importableRoot <> " {}; in toString (a.reflex.nixpkgs.lib.isDerivation a.passthru.staticFilesImpure))"
+          , "--raw"
+          -- `--raw` is not available with old nix-instantiate. It drops quotation
+          -- marks and trailing newline, so is very convenient for shelling out.
+          ]
+      -- Check whether the impure static files are a derivation (and so must be built)
+      readProcessAndLogStderr Debug $ if isDerivation == "1"
+        then proc "nix-build" ["-E", "(import " <> importableRoot <> "{}).passthru.staticFilesImpure"]
```

Comment:
Because no `--raw` for `nix build` or `nix-build`, I think we need to remove trailing newline